deeppavlov/dataset_readers/snips_reader.py

```diff
@@ -0,0 +1,83 @@
+# Copyright 2019 Neural Networks and Deep Learning lab, MIPT
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+from typing import List, Dict, Any, Optional
+from pathlib import Path
+
+from overrides import overrides
+
+from deeppavlov.core.common.registry import register
+from deeppavlov.core.data.dataset_reader import DatasetReader
+
+
+@register('snips_reader')
+class SnipsReader(DatasetReader):
+    """The class to read Snips NLU Benchmark dataset (custom intents section).
+
+    See https://github.com/snipsco/nlu-benchmark.
+    """
+
+    # noinspection PyAttributeOutsideInit
+    @overrides
+    def read(self, data_path: str, queries_per_intent: Optional[int] = None, test_validate_split: float = 0.5,
+             *args, **kwargs) -> \
+            Dict[str, List[Dict[str, Any]]]:
+        """
+        Each query in the output has the following form:
+            { 'intent': intent_name,
+              'data': [ { 'text': text, ('entity': slot_name)? } ]
+            }
+
+        Args:
+            data_path: A path to a folder with dataset files.
+            queries_per_intent: Number of queries to load for each intent. None to load all.
+                If the requested number is greater than available in file, all queries are returned.
+            test_validate_split: Proportion of `_validate` files to be used as test dataset (since Snips
+                is split into training and validation sets without a separate test set).
+        """
+        # TODO download the files automatically
```

Comment:
If the files are not downloaded automatically, there should be an error or a warning if the data is not there

Reply:
I'll just put the expected intents list in, then an error should be raised from trying to read a non-existent file.

Reply:
This `TODO` can be removed, it seems.